src/FSharp.Markdown/Markdown.fs

```diff
@@ -28,19 +29,19 @@ type MarkdownColumnAlignment =
 /// Represents inline formatting inside a paragraph. This can be literal (with text), various
 /// formattings (string, emphasis, etc.), hyperlinks, images, inline maths etc.
 type MarkdownSpan =
-  | Literal of string
-  | InlineCode of string
-  | Strong of MarkdownSpans
-  | Emphasis of MarkdownSpans
-  | AnchorLink of string 
-  | DirectLink of MarkdownSpans * (string * option<string>)
-  | IndirectLink of MarkdownSpans * string * string
-  | DirectImage of string * (string * option<string>)
-  | IndirectImage of string * string * string
-  | HardLineBreak
-  | LatexInlineMath of string
-  | LatexDisplayMath of string
-  | EmbedSpans of MarkdownEmbedSpans
+  | Literal of text:string * range:MarkdownRange option
+  | InlineCode of code:string * range:MarkdownRange option
+  | Strong of spans:MarkdownSpans * range:MarkdownRange option
+  | Emphasis of spans:MarkdownSpans * range:MarkdownRange option
+  | AnchorLink of link:string * range:MarkdownRange option
+  | DirectLink of spans:MarkdownSpans * linkAndTitle:(string * option<string>) * range:MarkdownRange option
+  | IndirectLink of spans:MarkdownSpans * link:string * key:string * range:MarkdownRange option
+  | DirectImage of body:string * linkAndTitle:(string * option<string>) * range:MarkdownRange option
```

Comment:
I would probably prefer to split `linkAndTitle` into two: `link:string * title:option<string>`